### PR TITLE
Fixes #245: NextGen dynamic thumbnails

### LINKED ISSUE #245: Add support for NextGen dynamic thumbnails
When displaying a gallery with thumbnails, those thumbnails are created dynamically.
[Source](https://app.intercom.io/a/apps/cd6nxj3z/inbox/inbox/conversation/15591324559).

### PR DESCRIPTION
**Pull request by Screenfeed**
Added process to optimize thumbnails that are created dynamically by NextGen Gallery.

### CODE REVIEW

File: inc/3rd-party/nextgen-gallery/inc/classes/class-imagify-ngg-attachment.php [hunk 1]
```diff
@@ -56,23 +65,34 @@ class Imagify_NGG_Attachment extends Imagify_Attachment {
 	 */
 	public function __construct( $id ) {
 		if ( is_object( $id ) ) {
-			$this->image = $id;
-			$this->id    = (int) $id->pid;
+			if ( is_a( $id, 'nggImage' ) ) {
```
Comment (Tabrisrp) [inline inc/3rd-party/nextgen-gallery/inc/classes/class-imagify-ngg-attachment.php]: You could use `instanceOf` insteand of `is_a()` which is somewhat more performant

`$id instanceOf nggImage`

File: inc/3rd-party/nextgen-gallery/nextgen-gallery.php [hunk 2]
```diff
@@ -19,6 +19,7 @@
 
 Imagify_NGG::get_instance()->init();
 Imagify_NGG_DB::get_instance()->init();
+Imagify_NGG_Dynamic_Thumbnails_Background_Process::get_instance();
```
Comment (Tabrisrp) [inline inc/3rd-party/nextgen-gallery/nextgen-gallery.php]: what's the use of this?

File: inc/3rd-party/nextgen-gallery/inc/classes/class-imagify-ngg-attachment.php (hunk 1 above)
Comment (Screenfeed) [reply inc/3rd-party/nextgen-gallery/inc/classes/class-imagify-ngg-attachment.php]: OK I didn't know about that, I'll do it.

File: inc/3rd-party/nextgen-gallery/nextgen-gallery.php (hunk 2 above)
Comment (Screenfeed) [reply inc/3rd-party/nextgen-gallery/nextgen-gallery.php]: Both `WP_Background_Process` and `WP_Async_Request` launch hooks when instantiated.

Comment (Tabrisrp) [reply inc/3rd-party/nextgen-gallery/nextgen-gallery.php]: the instanciation in `imagify_ngg_maybe_add_dynamic_thumbnail_to_background_process` is not enough?

Comment (Screenfeed) [reply inc/3rd-party/nextgen-gallery/nextgen-gallery.php]: No, they launch ajax callbacks and cron schedules, this must be done all the time.

**Pull request merged**